pkg/logs/input/kubernetes/launcher.go

```diff
@@ -220,9 +222,23 @@ func (l *Launcher) getSourceName(pod *kubelet.Pod, container kubelet.ContainerSt
 	return fmt.Sprintf("%s/%s/%s", pod.Metadata.Namespace, pod.Metadata.Name, container.Name)
 }
 
-// getPath returns the path where all the logs of the container of the pod are stored.
-func (l *Launcher) getPath(pod *kubelet.Pod, container kubelet.ContainerStatus) string {
-	return fmt.Sprintf("%s/%s/%s/*.log", podsDirectoryPath, pod.Metadata.UID, container.Name)
+// getPath returns a wildcard matching with any logs file of container in pod.
+func (l *Launcher) getPath(basePath string, pod *kubelet.Pod, container kubelet.ContainerStatus) string {
+	// the pattern for container logs is different depending on the version of Kubernetes
+	// so we need to try both format,
+	// previously it was `/var/log/pods/{pod_uid}/{container_name}/{n}.log`,
+	// then it changed to `/var/log/pods/{pod_namespace}_{pod_name}_{pod_uid}/{container_name}/{n}.log`.
+	// see: https://github.com/kubernetes/kubernetes/pull/74441 for more information.
+	oldDirectory := filepath.Join(basePath, pod.Metadata.UID)
+	if _, err := os.Stat(oldDirectory); err == nil {
+		return filepath.Join(basePath, pod.Metadata.UID, container.Name, anyLogFile)
+	}
+	return filepath.Join(basePath, l.getPodDirectory(pod), container.Name, anyLogFile)
+}
+
+// getPodDirectory returns the name of the directory of pod containers.
+func (l *Launcher) getPodDirectory(pod *kubelet.Pod) string {
```

Comment:
Could be interesting to have the mention of the starting version of kubernetes to have this pattern in the function name:
```golang
func (l *Launcher) getPodDirectorySince1_14(pod *kubelet.Pod) string { ... }
func (l *Launcher) getPodDirectoryUntil1_13(pod *kubelet.Pod) string { ... }
```